pkg/api/index.go

```diff
@@ -23,6 +23,22 @@ import (
 	"github.com/grafana/grafana/pkg/setting"
 )
 
+type URLPrefs struct {
+	Language string
+	Theme    string
+}
+
+// URL prefs take precedence over any saved user preferences
+func getURLPrefs(c *contextmodel.ReqContext) URLPrefs {
+	language := c.Query("lang")
+	theme := c.Query("theme")
```

Comment:
We should take locale as well I think.

Reply:
yeah i was sorting waiting for it all to settle, but i can add it now 👍